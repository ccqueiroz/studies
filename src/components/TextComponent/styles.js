import styled, { css } from 'styled-components';

const transformTextEllipsis = (textEllipsis) => {
  return textEllipsis
    ? css`
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    `
    : css`
      text-overflow: unset;
      white-space: unset;
      overflow: unset;
    `;
};

export const Container = styled.p`
  ${({
    theme,
    lineHeight,
    letterSpacing,
    textIndent,
    textAlign,
    wordBreak,
    textEllipsis,
  }) => css`
    font-size: ${theme.font.sizes.medium};
    line-height: ${lineHeight};
    letter-spacing: ${letterSpacing};
    text-indent: ${textIndent};
    text-align: ${textAlign};
    word-break: ${wordBreak ? 'break-word' : 'unset'};
    ${transformTextEllipsis(textEllipsis)};
  `}
`;
