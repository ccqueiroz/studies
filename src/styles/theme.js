export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
  },
  font: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      xsmall: '0.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
    letterSpacing: {
      none: 'unset',
      xsmall: '0.1rem',
      small: '0.2rem',
      medium: '0.3rem',
      large: '0.5rem',
      xlarge: '0.75rem',
      huge: '0.9rem',
      xhuge: '1rem',
    },
  },
  media: {
    ltMedium: '(max-width: 768px)',
  },
};
