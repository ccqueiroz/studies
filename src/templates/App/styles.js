import style, { css } from 'styled-components';

export const Wrapper = style.div`
  ${(props) => css`
    background: ${props.background};
  `}
`;
