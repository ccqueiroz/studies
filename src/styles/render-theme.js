import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderThemeProvider = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};
