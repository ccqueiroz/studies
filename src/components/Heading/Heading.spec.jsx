import { screen } from '@testing-library/react';
import { Heading } from '.';
import { renderThemeProvider } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('[Component]: <Heading />', () => {
  it('Should be render with default values', () => {
    renderThemeProvider(<Heading>Texto de Teste</Heading>);
    const heading = screen.getByRole('heading', {
      name: 'Texto de Teste',
    });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.font.sizes.xhuge,
      'text-transform': 'none',
    });
  });

  it('Should be render with white color', () => {
    renderThemeProvider(<Heading colorDark={false}>Texto de Teste</Heading>);
    const heading = screen.getByRole('heading', {
      name: 'Texto de Teste',
    });
    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('Should be render correctly heading small sizes', () => {
    renderThemeProvider(<Heading size="small">Texto de Teste</Heading>);
    const heading = screen.getByRole('heading', {
      name: 'Texto de Teste',
    });
    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.medium,
    });
  });

  it('Should be render correctly heading medium sizes', () => {
    renderThemeProvider(<Heading size="medium">Texto de Teste</Heading>);
    const heading = screen.getByRole('heading', {
      name: 'Texto de Teste',
    });
    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.large,
    });
  });

  it('Should be render correctly heading big sizes', () => {
    renderThemeProvider(<Heading size="big">Texto de Teste</Heading>);
    const heading = screen.getByRole('heading', {
      name: 'Texto de Teste',
    });
    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.xlarge,
    });
  });

  it('Should be render correctly uppercase letters', () => {
    renderThemeProvider(<Heading uppercase>Texto de Teste</Heading>);
    const heading = screen.getByRole('heading', {
      name: 'Texto de Teste',
    });
    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('Should be render correctly font-size when screen less then 768px', () => {
    renderThemeProvider(<Heading>Texto de Teste</Heading>);
    const heading = screen.getByRole('heading', {
      name: 'Texto de Teste',
    });

    expect(heading).toHaveStyleRule('font-size', theme.font.sizes.xlarge, {
      media: theme.media.ltMedium,
    });
  });

  it('Should be render correctly heading element', () => {
    const { container } = renderThemeProvider(
      <Heading as="h6">Texto de Teste</Heading>,
    );

    const h6 = container.querySelector('h6');

    expect(h6.tagName.toLowerCase()).toBe('h6');
  });
});
