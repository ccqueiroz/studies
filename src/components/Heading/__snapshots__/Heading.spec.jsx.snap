// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: <Heading /> Should be created snapshot component 1`] = `
.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<div>
  <h1
    class="c0"
  >
    Texto de Teste
  </h1>
</div>
`;
